### ISSUE TITLE
Removing associations in a nested structure

### ISSUE THREAD

**Issue by fwielstra** [go-gorm/gorm#3653]
## Your Question

I have a somewhat deeply nested structure that I'm trying to treat as a full document. One part of this structure is a list of references to another existing model.

Adding entries to this model works fine; I found the `FullSaveAssociations` option a little while ago, and I see it's been added to the docs recently as well.

However, removing it does not seem to work intuitively.

How can I best approach this? I have the outline of what I'm trying to do here: https://play.golang.org/p/TcTJ70JAtx5 ; I suspect I need to do something else or extra on line 85.

Further down I have an attempt that seemed to work elsewhere, using .Replace to fully swap out `Flow`s, but this causes an error "row value misused"

Is there a "definitive" way to use gorm in such a way that I can say "save this Flow", and it will handle everything automatically? Mostly what I'm stuck at now is updating, adding & removing any relationships. On a superficial level, using a combination of `FullSaveAssociations` and `Association().Replace()` things seem to work, but I don't feel confident at all in it.

**Comment by jinzhu** [comment]
>  On a superficial level, using a combination of FullSaveAssociations and Association().Replace() things seem to work, but I don't feel confident at all in it.

Yes, this will works.

`Association().Replace()` doesn't works with nested associations, it can only works with the first level.

**Comment by fwielstra** [comment]
I have made a working version; I iterate over the relationship and fetch each one. I then first delete all associations, then use `Replace()` to re-apply them:

https://play.golang.org/p/Qox6eWj4nDP

I tried to use `db.Model(&config).Association("Flows").Clear()`, but that generates a query that attempts to set `config_id` to`null`, which is disallowed because of the `not null` constraint.

I tried to call `.Append()` after the delete, but in my own code that seems to cause 5 entries to be appended (in my own code; cannot seem to reproduce it in the code).

There is one issue: sqlite gives an error or warning `row value misused` on line 91:

```
2020/10/22 11:25:14 scratch_44.go:91 row value misused
[0.161ms] [rows:0] UPDATE `config_flows` SET `config_id`=NULL WHERE (`config_flows`.`config_id`,`config_flows`.`flow_id`) NOT IN ((1,2),(1,3)) AND `config_flows`.`config_id` = 1
```
